identity.hhs

```diff
@@ -0,0 +1,38 @@
+/**
+ * @author Mudroad White
+ * @param input - a positive integer denoting the size of the expected matrix.
+ * @returns - a Mat object representing a 2D identity matrix.
+ */
+
+
+function identity(input) {
+
+	*import math: is_number
+
+	// Corner cases for input
+	// First, input should be a positive integer
+	if (!Number.isInteger(input) || input < 0) {
+		throw new Error('Expecped positive integer input')
+	}
+
+	// Then we return special Mat value when input is 0 or 1
+	if (input == 0) {
+		return new Mat([]);
+	}
+
+	if (input == 1) {
+		return new Mat([[1]]);
+	}
+
+	// Now we can start on creating the Id matrix
+	let result = [];
+	for (let i = 0; i < n; i++){
+		// For each row, we create a new Array object and fill the corresponded position with 1
+		let row = new Array(len).fill(0);
+		row[i] = 1;
+		result += row;
+	}
+
+	// return as a Mat object
+	return new Mat(result);
+}
```

Comment:
1. Create an N-by-N matrix instead of creating an N-by-N array and use constructure of Matrix, which copy the data twice;
2. Unit test is needed.